pkg/plugin/interfaces.go

```diff
@@ -53,7 +54,7 @@ type Subcommand interface {
 	// command line flags.
 	BindFlags(*pflag.FlagSet)
 	// Run runs the subcommand.
-	Run() error
+	Run(fs afero.Fs) error
```

Comment:
I'm still not convinced `afero.Fs` should be exposed in the signature. How about
```suggestion
	Run(RunOptions) error
```
Where
```go
type RunOptions struct {
	FS afero.Fs
}
```
So fields can be easily deprecated/added without changing the interface itself?

Reply:
Being `RunOptions` and exported type, and `FS` an exported field, we would still expose `afero.Fs`, right?

Reply:
Yes, but you could deprecate that field if it was later decided that another fs abstraction should be used without causing a breaking change.

Reply:
I suppose you could add parameters and not remove `afero.Fs` as to not make a breaking change, but that's uglier.